addons/uom/models/uom_uom.py

```diff
@@ -23,7 +23,7 @@ def _onchange_uom_ids(self):
             self.uom_ids[0].uom_type = 'reference'
             self.uom_ids[0].factor = 1
         else:
-            reference_count = sum(uom.uom_type == 'reference' for uom in self.uom_ids)
+            reference_count = sum(uom.uom_type == 'reference' for uom in self.uom_ids or self._origin.uom_ids)
             if reference_count == 0 and self._origin.id:
```

Comment:
@Sehnde Nice thank you, it works. I test on the interface user can leave the `uom_ids` empty is that ok right ?
Also testing when adding 2 line of uom then try to edit the reference line to smaller and bigger it will raise error as expected